### PR TITLE
Init non-bootstrap stage0-posix tools

### PR DESCRIPTION
**Pull request by emilytrau**
## Description of changes

Non-bootstrapped packages of stage0-posix intended for external use/prototyping

Related #227914

## Things done

- Built on platform(s)
  - [x] i686-linux
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] riscv32-linux
  - [ ] riscv64-linux
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/me/mescc-tools/package.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "mescc-tools";
+  version = "unstable-2023-05-23";
+
+  src = fetchgit {
+    url = "https://git.savannah.gnu.org/git/mescc-tools.git";
+    rev = "632357f807b7a5e518d1df368d465aebc7b4d139";
+    hash = "sha256-jFDrmzsjKEQKOKlsch1ceWtzUhoJAJVyHjXGVhjE9/U=";
```
Comment (wegank) [inline pkgs/by-name/me/mescc-tools/package.nix:15]:
```suggestion
    hash = "sha256-vDgafhwxy5dNyiPVawlzbrrDIWsbXvd6x2CqN/ME+fc=";
```

Comment (emilytrau) [reply pkgs/by-name/me/mescc-tools/package.nix:15]: For some reason this hash doesn't seem to be stable? `fetchSubmodules` is meant to be false but sometimes it tries anyways? But when I rebuilt the problem went away :/

Comment (emilytrau) [reply pkgs/by-name/me/mescc-tools/package.nix:15]: i didn't realise `fetchSubmodules` defaults to `true`. i've changed it to force `false`

**Review by wegank — APPROVED** [review]

File: pkgs/by-name/me/mescc-tools/package.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+, fetchgit
+, m2libc
+, which
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "mescc-tools";
+  version = "unstable-2023-05-23";
+
+  src = fetchgit {
```
Comment (picnoir) [inline pkgs/by-name/me/mescc-tools/package.nix:12]: You can use the savannah fetcher, it'll save you from downloading the git history:

```diff
diff --git a/pkgs/by-name/me/mescc-tools/package.nix b/pkgs/by-name/me/mescc-tools/package.nix
index 2c87ec13df39..dfaf152754a9 100644
--- a/pkgs/by-name/me/mescc-tools/package.nix
+++ b/pkgs/by-name/me/mescc-tools/package.nix
@@ -1,6 +1,6 @@
 { lib
 , stdenv
-, fetchgit
+, fetchFromSavannah
 , m2libc
 , which
 }:
@@ -9,11 +9,10 @@ stdenv.mkDerivation (finalAttrs: {
   pname = "mescc-tools";
   version = "unstable-2023-05-23";
 
-  src = fetchgit {
-    url = "https://git.savannah.gnu.org/git/mescc-tools.git";
+  src = fetchFromSavannah {
+    repo = "mescc-tools";
     rev = "632357f807b7a5e518d1df368d465aebc7b4d139";
     hash = "sha256-jFDrmzsjKEQKOKlsch1ceWtzUhoJAJVyHjXGVhjE9/U=";
-    fetchSubmodules = false;
   };
 
   # Don't use vendored M2libc
```

File: pkgs/by-name/m2/m2-planet/package.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    rev = "6232b03a53a60ce9c42776433888dfc26c42de28";
+    hash = "sha256-MAbvym7EaSMmWit44Qc+VnoolFf3x26PPXVvNhmo8io=";
+  };
+
+  # Don't use vendored M2libc
+  postPatch = ''
+    rmdir M2libc
+    ln -s ${m2libc}/include/M2libc M2libc
+  '';
+
```
Comment (picnoir) [inline pkgs/by-name/m2/m2-planet/package.nix:23]: Let's add a comment here:


```suggestion
# Upstream overrides the optimisation to be -O0, which is incompatible with fortify. Let's disable it.
```

**Review by picnoir — APPROVED** [review]
LGTM aside from two minor nits. Nits == fine not to address if you don't feel like it.

Taking advantage of this issue to thank you for all the work you're putting into bootstrapping :heart:  This is super valuable! Thanks for doing this.

Don't hesitate to ping me if you need reviews in the future.

**Comment by emilytrau** [comment]
@picnoir Thanks for the kudos and review. Appreciate it!

**Pull request merged**